src/components/Identity/createComponent.js

```diff
@@ -17,135 +17,135 @@ const addEcidToPayload = (payload, ecid) => {
   });
 };
 
-export default (processIdSyncs, config, logger, optIn, eventManager) => {
+export default (processIdSyncs, config, logger, consent, eventManager) => {
   const { idMigrationEnabled, orgId } = config;
   const identityCookieName = getNamespacedCookieName(
     orgId,
     IDENTITY_COOKIE_KEY
   );
   let deferredForEcid;
-  const migration = createMigration(orgId);
+  const migration = createMigration({ orgId, consent });
   const hasIdentityCookie = () => Boolean(cookieJar.get(identityCookieName));
-  const customerIds = createCustomerIds(eventManager);
+  const customerIds = createCustomerIds({ eventManager, consent });
+
+  // TO-DOCUMENT: We wait for ECID before trigger any events.
+  const accommodateIdentityOnRequest = payload => {
```

Comment:
Identities are passed on both consent requests and data collection requests. The first request of either type should pause subsequent requests. For this reason, I broke this logic out and called it from both `onBeforeDataCollectionRequest` and `onBeforeConsentRequest`.